hangman.rb

```diff
@@ -0,0 +1,49 @@
+
+$random_words = ["ruby", "hangman", "flutter", "developer", "omkar", "github", "josh", "assignment"]
+
+def choose_word
+  $random_words.sample
+end
+
+def display_word(word,correct_guesses)
+  word.chars.map { |char| correct_guesses.include?(char) ? char : '_' }.join(' ')
+end
+
+def won?(word,correct_guesses)
+  word.chars.all? { |char| correct_guesses.include?(char) }
+end  
+
+def play_hangman
+  word = choose_word
+  chances = word.size
+  correct_guesses = []
+  incorrect_guesses = []
+  
+  puts 'Welcome to Hangman'
+
+  until incorrect_guesses.size >= chances || won?(word,correct_guesses)
+    puts "Word: #{display_word(word,correct_guesses)}"
+    puts "Incorrect guesses: #{incorrect_guesses.join{""}}"
+    puts "Chances left: #{chances-incorrect_guesses.size}"
+    puts 'Enter a letter'
+    guess_letter = gets.chomp.downcase
+
+    if guess_letter.length!=1 || !('a'..'z').include?(guess_letter)
```

Comment:
Use the guard clause, instead of the if-else ladder
Also, try to move conditional statements to separate methods, it will make code readable and quick to understand.